graphqlapi/src/main/java/io/stargate/graphql/web/resources/schemafirst/FilesResource.java

```diff
@@ -38,9 +70,73 @@ public Response getCqlDirectives() {
         .build();
   }
 
+  @GET
+  @Path("/namespace/{namespaceName}.graphql")
+  @Produces(MediaType.TEXT_PLAIN)
+  public Response getSchema(
+      @HeaderParam("X-Cassandra-Token") String token,
+      @PathParam("namespaceName") String namespace,
+      @QueryParam("version") String version,
+      @Context HttpServletRequest httpRequest)
+      throws Exception {
+
+    SchemaSource schemaSource =
+        isAuthorized(token, namespace, httpRequest)
+            ? schemaSourceDao.getByVersion(
+                namespace, Optional.ofNullable(version).map(UUID::fromString))
+            : null;
+    if (schemaSource == null) {
+      return Response.status(
+              Response.Status.NOT_FOUND.getStatusCode(),
+              String.format(
+                  "The schema for namespace %s and version %s does not exists, "
+                      + "or you're not authorized to read it.",
```

Comment:
Not authorized should be a 401

Reply:
Do we want to make that distinction though? If we return a 401, we're leaking the fact that the namespace and version exist.